test/conformance/device_code/CMakeLists.txt

```diff
@@ -9,9 +9,15 @@ macro(add_device_binary SOURCE_FILE)
     file(MAKE_DIRECTORY ${DEVICE_BINARY_DIR})
     foreach(TRIPLE ${TARGET_TRIPLES})
         set(EXE_PATH "${DEVICE_BINARY_DIR}/${KERNEL_NAME}_${TRIPLE}")
+        if(${TRIPLE} MATCHES "amd")
+            set(AMD_TARGET_BACKEND -Xsycl-target-backend=${TRIPLE})
+            set(AMD_OFFLOAD_ARCH  --offload-arch=${AMD_ARCH})
+        endif()
         add_custom_command(OUTPUT ${EXE_PATH}
-            COMMAND ${UR_DPCXX} -fsycl -fsycl-targets=${TRIPLE} -fsycl-device-code-split=off
-            ${SOURCE_FILE} -o ${EXE_PATH}
+            COMMAND ${UR_DPCXX} -fsycl -fsycl-targets=${TRIPLE} -fsycl-device-code-split=off 
+            ${AMD_TARGET_BACKEND} ${AMD_OFFLOAD_ARCH} ${SOURCE_FILE} 
+            -o ${EXE_PATH} || (exit 0)
```

Comment:
Ignoring the error on dumping the device binaries already makes it easy to accidentally misconfigure these tests. Ignoring compiler errors will only make the problem worse. Can you please add a comment why the errors are ignored?

I'm wondering whether it would be possible to check if running `${EXE_PATH}` created a file in cwd. Maybe with inotify?

Reply:
This is a slightly strange issue I've found. AMD does not support images (at least through SYCL) and so the compilation of `image_copy.cpp` fails. I was using Ninja as a generator locally and it's perfectly happy to ignore this error and let the build continue, but using "Unix Makefiles" stops the build and exits. Perhaps we should just skip compiling this code on AMD and avoid having to silence the error. I've already added checks to the image/sampler CTS to ensure that this program is not loaded on AMD. 👍